packages/core/src/behavior.ts

```diff
@@ -135,7 +140,9 @@ export function createPromiseBehavior<T, TEvent extends EventObject>(
     receiveSignal: (actorContext: ActorContext, signal: LifecycleSignal) => {
       switch (signal) {
         case startSignal:
-          const resolvedPromise = Promise.resolve(promise);
+          const resolvedPromise = Promise.resolve(
```

Comment:
should we allow for anything other than machines to be "eager"? with machines we can accept plain machines - rather than enforcing machine creators at all times as we know that machines are stateless and thus it's safe

related question - should `createXBehavior` create the very same "template" at all times for a given set of arguments? if yes then we would have to require entity creators (lazy) as arguments to achieve this, if no then I'm not sure how useful accepting lazy creators is here - this is not the main user-facing API and those lazy creators get resolved in `invokeX` already

Reply:
Yeah, that's a change I'm planning to make in a separate PR - we should enforce that all spawned actors be lazy, e.g.:

```diff
-spawn(promise);
+spawn(() => promise);
```

Reply:
👍